src/routes/Profile/WorkHistory/modals/EditToimenkuvaModal.tsx

```diff
@@ -0,0 +1,253 @@
+  onClose: () => void;
+  tyopaikkaId: string;
+  toimenkuvaId: string;
+}
+
+interface ToimenkuvaForm {
+  id: string;
+  nimi: string;
+  kuvaus: string;
+  alkuPvm: string;
+  loppuPvm: string;
+  osaamiset: { id: string; nimi: string }[];
+}
+
+const MainStep = () => {
+  const { t } = useTranslation();
+  const { register, control } = useFormContext<ToimenkuvaForm>();
+  return (
+    <>
+      <h2 className="mb-4 text-heading-3 text-black sm:mb-5 sm:text-heading-2">
+        {t('work-history.edit-job-description')}
+      </h2>
+      <div className="mb-6">
+        <InputField
+          label={t('work-history.job-description')}
+          {...register('nimi')}
+          placeholder="Lorem ipsum dolor sit amet"
+        />
+      </div>
+      <div className="mb-6 flex grow gap-6">
+        <div className="block w-full">
+          <Controller
+            control={control}
+            render={({ field }) => (
+              <Datepicker label={t('work-history.started')} {...field} placeholder={t('date-placeholder')} />
+            )}
+            name="alkuPvm"
+          />
+        </div>
+        <div className="block w-full">
+          <Controller
+            control={control}
+            render={({ field }) => (
+              <Datepicker label={t('work-history.ended')} {...field} placeholder={t('date-or-continues-placeholder')} />
+            )}
+            name="loppuPvm"
+          />
+        </div>
+      </div>
+    </>
+  );
+};
+
+const OsaaminenStep = () => {
+  const [debouncedDescription, description, setDescription] = useDebounceState('', 500);
+  const { t } = useTranslation();
+  const { control } = useFormContext<ToimenkuvaForm>();
+  return (
+    <>
+      <h2 className="mb-4 text-heading-3 text-black sm:mb-5 sm:text-heading-2">{t('work-history.edit-competences')}</h2>
+      <div className="mb-6">
+        <InputField
+          label={t('work-history.edit-competences')}
+          value={description}
+          onChange={(event: React.ChangeEvent<HTMLInputElement>) => setDescription(event.target.value)}
+          placeholder="Lorem ipsum dolor sit amet"
+        />
+      </div>
+
+      <Controller
+        control={control}
+        name="osaamiset"
+        render={({ field: { onChange, value } }) => (
+          <OsaamisSuosittelija description={debouncedDescription} onChange={onChange} value={value} />
+        )}
+      />
+    </>
+  );
+};
+
+const EditToimenkuvaModal = ({ isOpen, onClose, tyopaikkaId: id, toimenkuvaId }: EditToimenkuvaModalProps) => {
+  const deleteKoulutukset = async () => {
+    await client.DELETE('/api/profiili/tyopaikat/{id}/toimenkuvat/{toimenkuvaId}', {
+      params: { path: { id, toimenkuvaId } },
+    });
+
+    onClose();
+  };
+
+  const {
+    t,
+    i18n: { language },
+  } = useTranslation();
+
+  if (!id) {
+    onClose();
+  }
+
+  const formId = React.useId();
+  const [step, setStep] = React.useState(0);
+  const steps = [MainStep, OsaaminenStep];
```

Comment:
Not sure, but it might be beneficial to have this `steps` to have different name. Maybe `stepComponents` or something?

It got me distracted at first when just line above the singular step means number, but this plural means UI components.

Reply:
Renamed